src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java

```diff
@@ -1119,15 +1112,18 @@ public ExamChecklistDTO getStatsForChecklist(Exam exam, boolean isInstructor) {
                 log.debug("StatsTimeLog: number of student exams submitted done in {}", TimeLogUtil.formatDurationFrom(start));
             }
 
-            examChecklistDTO.setNumberOfExamsStarted(numberOfStudentExamsStarted);
-            examChecklistDTO.setNumberOfExamsSubmitted(numberOfStudentExamsSubmitted);
+            return new ExamChecklistDTO(numberOfGeneratedStudentExams, numberOfTestRuns, totalNumberOfAssessmentsFinished, totalNumberOfParticipationsForAssessment,
+                    numberOfStudentExamsSubmitted, numberOfStudentExamsStarted, totalNumberOfComplaints, totalNumberOfComplaintResponse, exercisesPrepared, null, null);
+
         }
-        examChecklistDTO.setNumberOfTotalParticipationsForAssessment(totalNumberOfParticipationsForAssessment);
+
         boolean existsUnassessedQuizzes = submissionRepository.existsUnassessedQuizzesByExamId(exam.getId());
-        examChecklistDTO.setExistsUnassessedQuizzes(existsUnassessedQuizzes);
         boolean existsUnsubmittedExercises = submissionRepository.existsUnsubmittedExercisesByExamId(exam.getId());
-        examChecklistDTO.setExistsUnsubmittedExercises(existsUnsubmittedExercises);
-        return examChecklistDTO;
+
+        // For non-instructors, consider what limited information they should receive and adjust accordingly
+        return new ExamChecklistDTO(null, null, totalNumberOfAssessmentsFinished, totalNumberOfParticipationsForAssessment, null, null, null, null, null, existsUnassessedQuizzes,
```

Comment:
would it make sense to add another constructor/method to the DTO that only takes the data instructors should see as input?

Reply:
Then maybe it makes sense to mark those values as `@Nullable` in the DTO? (use the jakarta Nullable)